Library/Homebrew/brew.rb

```diff
@@ -32,7 +32,7 @@
 
   empty_argv = ARGV.empty?
   help_flag_list = %w[-h --help --usage -?]
-  help_flag = !ENV["HOMEBREW_HELP"].nil?
```

Comment:
This environment variable should not be ignored.

Reply:
The problem is, if the help request needs a `brew tap` before fulfilling it (as in the example above), it detects this variable and outputs the `tap` help instead of actually tapping the cask.

Reply:
Sure but in that case handle the `brew tap` case more specifically either be unsetting `HOMEBREW_HELP` in `brew.sh` when appropriate (https://github.com/Homebrew/brew/blob/cf349018d5cc508ddd2bd0797df47688b9f7346b/Library/Homebrew/brew.sh#L211-L218) or just for the `brew tap` here: https://github.com/Homebrew/brew/blob/cf349018d5cc508ddd2bd0797df47688b9f7346b/Library/Homebrew/brew.rb#L119-L120

Reply:
Done, see 0998e16.